addons/website/models/res_lang.py

```diff
@@ -16,10 +16,15 @@ def write(self, vals):
 
     @api.model
     @tools.ormcache_context(keys=("website_id",))
-    def get_available(self):
+    def get_frontend(self):
+        """ Return the available languages for current request """
         if request and getattr(request, 'is_frontend', True):
-            return self.env['website'].get_current_website().language_ids.get_sorted()
-        return super().get_available()
+            return {
+                code: info
+                for code, info in self.get_installed_info_by('code').items()
+                if info.id in self.env['website'].get_current_website().language_ids.ids
+            }
```

Comment:
Not sure about this, but wouldn't it be worth factoring out `self.env['website'].get_current_website().language_ids.ids` outside the iteration?